server/bleep/src/webserver/answer.rs

```diff
@@ -429,10 +434,16 @@ async fn handle_inner(
             }
             AnswerProgress::Search(rephrased_query) => {
                 // TODO: Clean up this query handling logic
-                let s = search_snippets(&semantic, &params.q, rephrased_query).await?;
-                info!("Retrieved {} snippets", s.len());
+                let all_snippets = search_snippets(&semantic, &params.q, rephrased_query).await?;
+                info!("Retrieved {} snippets", all_snippets.len());
+
+                event.write().await.stages.push(
+                    Stage::new("semantic_results", &all_snippets).with_time(stop_watch.lap()),
+                );
+
+                let filtered_snippets = deduplicate_snippets(all_snippets);
 
-                if s.is_empty() {
+                if filtered_snippets.is_empty() {
```

Comment:
What happens if `all_snippets` is empty? Will this trigger an error in `deduplicate_snippets`?

Reply:
`deduplicate_snippets` is expected to only remove duplicates, if an empty vector is passed in, and empty vector is returned.